Src/FluentAssertions/StringWithOccurenceConstraint.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Diagnostics;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+
+namespace FluentAssertions
+{
+    [DebuggerNonUserCode]
+    public class StringWithOccurenceConstraint
+    {
+        public delegate bool StringOccurrenceComparer(int occuredTimes, int expectedTimes);
+
+        public StringWithOccurenceConstraint(StringAssertions parentConstraint,
+            string expected,
+            int occurrences,
+            string mode,
+            StringOccurrenceComparer comparator)
+        {
+            Assertion = parentConstraint;
+            Expected = expected;
+            Occurrences = occurrences;
+            Mode = mode;
+            Comparator = comparator;
+        }
+
+        private StringAssertions Assertion { get; }
+        private string Expected { get; }
+        private int Occurrences { get; }
+        private string Mode { get; }
+        private StringOccurrenceComparer Comparator { get; }
+
+        public AndConstraint<StringAssertions> Once(string because = "", params object[] becauseArgs) =>
+            Times(1, "once", because, becauseArgs);
+
+        public AndConstraint<StringAssertions> Twice(string because = "", params object[] becauseArgs) =>
+            Times(2, "twice", because, becauseArgs);
+
+        public AndConstraint<StringAssertions> Thrice(string because = "", params object[] becauseArgs) =>
+            Times(3, "thrice", because, becauseArgs);
+
+        public AndConstraint<StringAssertions> Times(int times, string because = "", params object[] becauseArgs) =>
+            Times(times, $"{times} times", because, becauseArgs);
+
+        private AndConstraint<StringAssertions> Times(int times, string displayValue, string because = "",
+            params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .ForCondition(Comparator(Occurrences, times))
+                .BecauseOf(because, becauseArgs)
+                .FailWith(
+                    $"Expected {{context:string}} {{0}} to contain {{1}} {Mode} {displayValue}{{reason}}, but found {Occurrences} times.",
```

Comment:
❓ What if `Occurrences` is 1? Should it not say _time_ instead of _times_?

Reply:
Good point, I will fix it